source/MaterialXTest/MaterialXCore/Units.cpp

```diff
@@ -153,32 +152,31 @@ TEST_CASE("UnitDocument", "[units]")
             {
                 if (pNode->getInputCount()) {
                     for (mx::InputPtr input : pNode->getInputs()) {
-                        const mx::TypeDesc* type = mx::TypeDesc::get(input->getType());
+                        const std::string type = input->getType();
                         const mx::ValuePtr value = input->getValue();
                         if (input->hasUnit() && value) {
-
-                            if (type->isScalar() && type->getBaseType() == mx::TypeDesc::BASETYPE_FLOAT)
+                            if (type == "float")
                             {
                                 float originalval = value->asA<float>();
                                 float convertedValue = uconverter->convert(originalval, input->getUnit(), DISTANCE_DEFAULT);
                                 float reconvert = uconverter->convert(convertedValue, DISTANCE_DEFAULT, input->getUnit());
                                 REQUIRE((originalval - reconvert) < EPSILON);
                             }
-                            else if (type->isFloat2())
+                            else if (type == "vector2" || type == "color2")
```

Comment:
unit conversion is not allowed on color types so the color2, color3, color4 inclusion should be removed.

Reply:
Fixed